client/app/project/project.controller.js

```diff
@@ -1266,35 +1275,39 @@
          * Call api to lock currently selected task for validation.  Will update view and map after unlock.
          */
         vm.lockSelectedTaskValidation = function () {
-            vm.lockingReason = 'VALIDATION';
-            var projectId = vm.projectData.projectId;
-            var taskId = vm.selectedTaskData.taskId;
-            var taskIds = [taskId];
-            // - try to lock the task, call returns a promise
-            var lockPromise = taskService.lockTasksValidation(projectId, taskIds);
-            lockPromise.then(function (tasks) {
-                //TODO - The following reset lines are repeated in several places in this file.
-                //Refactoring to a single function call was considered, however it was decided that the ability to
-                //call the resets individually was desirable and would help readability.
-                //The downside is that any change will have to be replicated in several places.
-                //A fundamental refactor of this controller should be considered at some stage.
-                vm.resetErrors();
-                vm.resetStatusFlags();
-                vm.resetTaskData();
-                // refresh the project, to ensure we catch up with any status changes that have happened meantime
-                // on the server
-                refreshProject(projectId);
-                updateMappedTaskPerUser(projectId);
-                vm.currentTab = 'validation';
-                vm.validatingStep = 'locked';
-                vm.selectedTaskData = tasks[0];
-                vm.isSelectedValidatable = true;
-                vm.lockedTaskData = tasks[0];
-                vm.lockTime[taskId] = getLastLockedAction(vm.lockedTaskData).actionDate;
-                formatHistoryComments(vm.selectedTaskData.taskHistory);
-            }, function (error) {
-                onLockError(projectId, error);
-            });
+            if(vm.user.emailAddress != 'uwaiszaki104@gmail.com'){
```

Comment:
Hi @uwaiszaki,
Could you improve the code to make it available to any user. So you should verify here if the username email is not null or `''`.

Reply:
@willemarcel I have added the changes and also rebased this branch in the develop branch.